napari/_qt/_qapp_model/_tests/test_processors.py

```diff
@@ -18,3 +51,18 @@ def test_add_layer_data_to_viewer():
             return_type=Union[ImageData, LabelsData],
             viewer=v,
         )
+
+
+def test_add_layer_to_viewer():
+    layer = MagicMock()
+    viewer = MagicMock()
+    _add_layer_to_viewer(None)
+    _add_layer_to_viewer(layer)
```

Comment:
I think practically when using this, when a viewer is not provided, we'd want a provider to dependency inject one.

I appreciate that testing the dependency injection seems like we are testing beyond our project, but it would be nice to check everything works together and would allow us to check for changes in upstream (e.g., when magicgui stopped accepting both `list` and `List`, sorry I forget the issue for this). I'll see what @DragaDoncila thinks about this.

Reply:
Mmm yeah I think it would be good to have a couple of tests with `make_napari_viewer` and check that widgets/layers are added. Maybe even just in one test so that we only have to build one viewer, keep the test suite a little less bloated.

Reply:
Are you talking about `make_napari_viewer` and passing the viewer or letting to viewer be passed via dependency injection?

Reply:
ah sorry, I was talking about passing the viewer. I think it would be good to also have a test for the dependency injection, even though it's not really a unit test